src/extract.js

```diff
@@ -49,12 +49,15 @@ exports.TranslationReference = class TranslationReference {
   }
 };
 
-function preprocessScriptTags(data, type) {
+function preprocessVueFile(data) {
   const vueFile = vueCompiler.parse({ source: data, needMap: false });
-
-  return (type !== 'vue' || ! vueFile.script)
-    ? ''
-    : vueFile.script.content.trim();
+  if (!vueFile.script) {
+    return null;
+  }
+  return {
+    content: vueFile.script.content.trim(),
+    lang: vueFile.script.lang,
```

Comment:
I suggest (cf higher in the comments)

```suggestion
    lang: vueFile.script.lang || 'js',
```

to avoid the undefined by default